gk/main.c

```diff
@@ -278,6 +291,14 @@ gk_process_request(struct flow_entry *fe, struct ipacket *packet,
 			fe->u.request.last_packet_seen_at);
 	struct gk_fib *fib = fe->grantor_fib;
 	struct ether_cache *eth_cache;
+	struct gatekeeper_if *back = &sol_conf->net->back;
+	struct rte_mbuf *pkt = packet->pkt;
+
+	rte_prefetch0(fib);
+	rte_prefetch0(back);
+	prefetch0_128_bytes(rte_pktmbuf_mtod_offset(
+		pkt, void *, 0));
+	prefetch0_128_bytes(pkt);
```

Comment:
Could you evaluate the impact of each of these prefetches?

Reply:
Good points! Will do.

Reply:
This evaluation is better done with perf since it should present more pronounced numbers.

Given that you're running perf on the terminal, you can copy the text and paste on GitHub as code. This would keep the evaluation records here on GitHub. Not only does it simplify our communication, but it also keeps the evaluation traceable by someone reading this review in the future.

Reply:
Should we also evaluate all the 9 scenarios (i.e., 1, 2, 3 lcores and 2^15, 2^20, 2^22 table sizes) using perf?

Reply:
Let's limit the tests to 3 lcores and 2^22-entry flow tables.

Reply:
Sounds good! Will do.